[CN-1354]: add truststore info to tls doc (#234)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/tls.adoc

== Prerequisites for TLS

> Before you create and configure TLS, you need to create Kubernetes secret. The Operator uses the secret to create truststore and keystore, and it modifies the Hazelcast configuration according to it.

Reviewer: ```suggestion
Before you create and configure TLS, you need to create a Kubernetes secret. The Operator uses the secret to create a truststore and keystore, and it modifies the Hazelcast configuration accordingly.
```

Example:

[source,shell]
----
kubectl create secret generic example \
  --from-file=tls.crt=example.crt \
  --from-file=tls.key=example.key
----

`ca.crt` is optional. You can also provide it to create the truststore:

[source,shell]
----
kubectl create secret generic example \
  --from-file=tls.crt=example.crt \
  --from-file=tls.key=example.key \
  --from-file=ca.crt=ca.crt
----

> NOTE: If you provide `ca.crt`, it creates a truststore using it, if you don't create it, truststore and keystore will be the same.

Reviewer: ```suggestion
NOTE: If you provide `ca.crt`, it creates a truststore using it, if you don't create it, the truststore and keystore will be the same.
```

> NOTE: The Operator creates keystore and truststore in link:https://en.wikipedia.org/wiki/PKCS_12[PKCS12] format.

Reviewer: ```suggestion
NOTE: The Operator creates the keystore and truststore in link:https://en.wikipedia.org/wiki/PKCS_12[PKCS12] format.
```

== Prerequisites for TLS

Before you create and configure TLS, you need to create Kubernetes secret. Kubernetes secret can be created in two different ways:

If you just want to provide certificates to create the keystore:

> 

Reviewer 2: Could you provide more information about internal implementation to give some insights to users? So they can create their own solutions.

Author: https://github.com/hazelcast/hazelcast-platform-operator-docs/pull/234/commits/29d69be1ddfbc6eef110e906f8959f0d82588f28

[source,shell]
----

> kubectl create secret tls example --cert=example.crt --key=example.key

Reviewer 2: we should use `kubectl create secret generic` for all options to prevent confusion.

Author: https://github.com/hazelcast/hazelcast-platform-operator-docs/pull/234/commits/29d69be1ddfbc6eef110e906f8959f0d82588f28

----

> 

Reviewer 2: Could you add note for encrypted keys?

Author: https://github.com/hazelcast/hazelcast-platform-operator-docs/pull/234/commits/29d69be1ddfbc6eef110e906f8959f0d82588f28
